File: catalyst-gateway/bin/src/service/common/objects/cardano/network.rs
//! Defines API schemas of Cardano network types.

use poem_openapi::Enum;

/// Cardano network type.
#[derive(Clone, Enum, Debug)]
pub(crate) enum Network {
    /// Cardano mainnet.
    #[oai(rename = "mainnet")]
    Mainnet,
    /// Cardano preprod.
    #[oai(rename = "preprod")]
    Preprod,
    /// Cardano preview.
    #[oai(rename = "preview")]
    Preview,
}

impl From<Network> for cardano_chain_follower::Network {
    fn from(value: Network) -> Self {
        match value {
            Network::Mainnet => Self::Mainnet,
            Network::Preprod => Self::Preprod,
            Network::Preview => Self::Preview,
        }
    }
}

File: catalyst_voices/packages/internal/catalyst_voices_repositories/lib/src/api/models/network.dart
import 'package:json_annotation/json_annotation.dart';

/// Cardano network type.
///
/// Used as query parameter for GET /v1/cardano/assets/{stake_address}.
@JsonEnum(valueField: 'value')
enum Network {
  mainnet('mainnet'),
  preprod('preprod'),
  preview('preview');

  final String value;

  const Network(this.value);
}
